tests/core/string/test_string.h

```diff
@@ -348,22 +348,103 @@ TEST_CASE("[String] Substr") {
 	CHECK(s.substr(3) == "ler Baby");
 }
 
+#define MULTICHECK_STRING_EQ(obj, func, param1, eq) \
+	do {                                            \
+		CHECK(obj.func(param1) == eq);              \
+		CHECK(obj.func(String(param1)) == eq);      \
+	} while (0)
```

Comment:
Any reason for the `do/while`, copied from somewhere?

Reply:
It's a common practice to help scope the macros and use them as if they were regular statements or function calls. In this particular case  I suppose I could drop them -- `CHECK` probably has them once it's expanded due to how it's used. I don't mind removing the ones I added.

Reply:
The other checks here doesn't have them but just uses a simple empty block, I've never seen this particular practice though

Reply:
Here's a quick example in our code base -- keep an eye on `INIT_XR_FUNC_V` and `EXT_INIT_XR_FUNC_V`:

https://github.com/godotengine/godot/blob/5ee983188de97ae027f9b9c1443438063f708a7e/modules/openxr/util.h#L36-L43


https://github.com/godotengine/godot/blob/5ee983188de97ae027f9b9c1443438063f708a7e/modules/openxr/extensions/openxr_fb_passthrough_extension_wrapper.cpp#L219-L222

Reply:
I just tested mine without the `do/while` -- I'll drop them